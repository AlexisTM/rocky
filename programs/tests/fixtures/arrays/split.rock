Say "IN-PLACE"
Let the string be "abc"
Split the string
Say the string at 0
Say the string at 1
Say the string at 2
Say the string at 3

Say "LITERAL"
Split "wxyz" into digits
Say digits at 0
Say digits at 1
Say digits at 2
Say digits at 3

Say "CUT"
Cut "1.2" into numbers with "."
Say numbers at 0
Say numbers at 1

Say "SHATTER"
Shatter "192.168.0.1" into ipaddr with "."
Say ipaddr at 0
Say ipaddr at 1
Say ipaddr at 2
Say ipaddr at 3


Say "BACKSLASH"
Shatter "\\users\\gaborsch\\rocky" into path with "\\"
Say path at 0
Say path at 1
Say path at 2
Say path at 3

Say "STAR"
Shatter "1*2*3*4" into path with "*"
Say path at 0
Say path at 1
Say path at 2
Say path at 3