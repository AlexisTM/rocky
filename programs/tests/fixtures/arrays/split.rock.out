IN-PLACE
a
b
c
mysterious
LITERAL
w
x
y
z
CUT
1
2
SHATTER
192
168
0
1
BACKSLASH

users
gaborsch
rocky
STAR
1
2
3
4
